# HG changeset patch
# User Rik <rdrider0-list@yahoo.com>
# Date 1240601341 25200
#      Fri Apr 24 12:29:01 2009 -0700
# Node ID 5247e89688e13821d971b3c42a11af97b4c75114
# Parent  4c9aff0c9a619b63d28ac770c99aeae5fe366d16
Eliminate most overfull errors when running texi2pdf for generating pdf documentation

Use @smallexample when necessary to reduce font for long lines.
Reword variables or phrases so that Tex can break them at a better spot.

diff --git a/doc/ChangeLog b/doc/ChangeLog
--- a/doc/ChangeLog
+++ b/doc/ChangeLog
@@ -1,8 +1,15 @@
+2009-04-24  Rik  <rdrider0-list@yahoo.com>
+	* interpreter/octave.pdf: 
+	Eliminate most overfull errors when running texi2pdf for generating 
+	pdf documentation.
+	Use @smallexample when necessary to reduce font for long lines.
+	Reword variables or phrases so that Tex can break them at a better spot.
+
 2009-04-21  Rik  <rdrider0-list@yahoo.com>
 	* interpreter/octave.pdf: Eliminate 'unbalanced parentheses in @def...'
 	error during texi2pdf.  Fixed four .m files incorrectly using @deftypefn 
 	macro.
 
 2009-04-19  Ansgar Burchardt  <ansgar@43-1.org>
 
 	* interpreter/numbers.txi: Fix small mistake in example.
diff --git a/doc/interpreter/basics.txi b/doc/interpreter/basics.txi
--- a/doc/interpreter/basics.txi
+++ b/doc/interpreter/basics.txi
@@ -813,24 +813,23 @@ mistake in the previous function definit
 
 @example
 octave:13> f ()
 @end example
 
 @noindent
 Octave will respond with
 
-@c Using 'smallexample' to make text fit on page when creating smallbook.
-@smallexample
+@example
 @group
 error: `x' undefined near line 1 column 24
 error: called from:
 error:   f at line 1, column 22
 @end group
-@end smallexample
+@end example
 
 @noindent
 This error message has several parts, and gives quite a bit of
 information to help you locate the source of the error.  The messages
 are generated from the point of the innermost error, and provide a
 traceback of enclosing expressions and function calls.
 
 In the example above, the first line indicates that a variable named
diff --git a/doc/interpreter/expr.txi b/doc/interpreter/expr.txi
--- a/doc/interpreter/expr.txi
+++ b/doc/interpreter/expr.txi
@@ -934,21 +934,20 @@ b = s
 @end group
 @end example
 
 @noindent
 The number of values on the left side of the expression can, however,
 not exceed the number of values on the right side.  For example, the
 following will produce an error.
 
-@c Using 'smallexample' to make text fit on page when creating smallbook.
-@smallexample
+@example
 [a, b, c, d] = [u, s, v] = svd (a);
 @print{} error: element number 4 undefined in return list
-@end smallexample
+@end example
 
 @opindex +=
 A very common programming pattern is to increment an existing variable
 with a given value, like this
 
 @example
 a = a + 2;
 @end example
diff --git a/scripts/general/cplxpair.m b/scripts/general/cplxpair.m
--- a/scripts/general/cplxpair.m
+++ b/scripts/general/cplxpair.m
@@ -32,17 +32,17 @@
 ## dimension of @var{z}.  If @var{dim} is specified, then the complex
 ## pairs are sorted along this dimension.
 ##
 ## Signal an error if some complex numbers could not be paired.  Requires
 ## all complex numbers to be exact conjugates within @var{tol}, or signals an 
 ## error.  Note that there are no guarantees on the order of the returned
 ## pairs with identical real parts but differing imaginary parts.
 ##
-## @c Using 'smallexample' to make text fit in page when using 'smallbook'
+## @c Set example in small font to prevent overfull line
 ## @smallexample
 ## cplxpair (exp(2i*pi*[0:4]'/5)) == exp(2i*pi*[3; 2; 4; 1; 0]/5)
 ## @end smallexample
 ## @end deftypefn
 
 ## FIXME: subsort returned pairs by imaginary magnitude
 ## FIXME: Why doesn't exp(2i*pi*[0:4]'/5) produce exact conjugates. Does
 ## FIXME:    it in Matlab?  The reason is that complex pairs are supposed
diff --git a/scripts/miscellaneous/edit.m b/scripts/miscellaneous/edit.m
--- a/scripts/miscellaneous/edit.m
+++ b/scripts/miscellaneous/edit.m
@@ -82,19 +82,20 @@
 ## @item "gnudoit -q \"(find-file \\\"%s\\\")\""   
 ## Send it to current Emacs; must have @code{(gnuserv-start)} in @file{.emacs}.
 ## @end table
 ##
 ## See also field 'mode', which controls how the editor is run by Octave.
 ## 
 ## On Cygwin, you will need to convert the Cygwin path to a Windows
 ## path if you are using a native Windows editor.  For example
-## @example
+## @c Set example in small font to prevent overfull line
+## @smallexample
 ## '"C:/Program Files/Good Editor/Editor.exe" "$(cygpath -wa %s)"'
-## @end example
+## @end smallexample
 ##
 ## @item home
 ## This is the location of user local m-files.  Be be sure it is in your
 ## path.  The default is @file{~/octave}.
 ##
 ## @item author
 ## This is the name to put after the "## Author:" field of new functions.
 ## By default it guesses from the @code{gecos} field of password database.
diff --git a/scripts/miscellaneous/intwarning.m b/scripts/miscellaneous/intwarning.m
--- a/scripts/miscellaneous/intwarning.m
+++ b/scripts/miscellaneous/intwarning.m
@@ -25,16 +25,17 @@
 ##
 ## @table @asis
 ## @item "query"
 ## The state of the Octave integer conversion and math warnings is
 ## queried.  If there is no output argument, then the state is printed.
 ## Otherwise it is returned in a structure with the fields "identifier"
 ## and "state".
 ##
+## @c Set example in small font to prevent overfull line
 ## @smallexample
 ## @group
 ## intwarning ("query")
 ## The state of warning "Octave:int-convert-nan" is "off"
 ## The state of warning "Octave:int-convert-non-int-val" is "off"
 ## The state of warning "Octave:int-convert-overflow" is "off"
 ## The state of warning "Octave:int-math-overflow" is "off"
 ## @end group
diff --git a/scripts/optimization/fsolve.m b/scripts/optimization/fsolve.m
--- a/scripts/optimization/fsolve.m
+++ b/scripts/optimization/fsolve.m
@@ -87,17 +87,17 @@
 ## called with that vector, then the intermediate results should be saved for
 ## future jacobian evaluation, and should be kept until a jacobian evaluation
 ## is requested or until outputfcn is called with a different vector, in which
 ## case they should be dropped in favor of this most recent vector.  A short
 ## example how this can be achieved follows:
 ##
 ## @example
 ## @group
-## function [fvec, fjac] = my_optim_func (x, optimvalues, state)
+## function [fvec, fjac] = user_func (x, optimvalues, state)
 ## persistent sav = [], sav0 = [];
 ## if (nargin == 1)
 ##   ## evaluation call
 ##   if (nargout == 1)
 ##     sav0.x = x; # mark saved vector
 ##     ## calculate fvec, save results to sav0.
 ##   elseif (nargout == 2)
 ##     ## calculate fjac using sav.
@@ -108,17 +108,17 @@
 ##     sav = sav0;
 ##   endif
 ##   ## maybe output iteration status etc.
 ## endif
 ## endfunction
 ##
 ## ## @dots{}.
 ## 
-## fsolve (@@my_optim_func, x0, optimset ("OutputFcn", @@my_optim_func, @dots{}))
+## fsolve (@@user_func, x0, optimset ("OutputFcn", @@user_func, @dots{}))
 ## @end group
 ## @end example
 ###
 ## @end deftypefn
 
 ## PKG_ADD: __all_opts__ ("fsolve");
 
 function [x, fvec, info, output, fjac] = fsolve (fcn, x0, options = struct ())
diff --git a/scripts/pkg/pkg.m b/scripts/pkg/pkg.m
--- a/scripts/pkg/pkg.m
+++ b/scripts/pkg/pkg.m
@@ -138,17 +138,17 @@
 ## @example
 ## p = pkg prefix
 ## @end example
 ##
 ## The location in which to install the architecture dependent files can be
 ## independent specified with an addition argument.  For example
 ##
 ## @example
-## pkg prefix ~/my_octave_packages ~/my_octave_packages_for_my_pc
+## pkg prefix ~/my_octave_packages ~/my_arch_dep_pkgs
 ## @end example
 ## @item local_list
 ## Set the file in which to look for information on the locally
 ## installed packages.  Locally installed packages are those that are
 ## typically available only to the current user.  For example
 ## @example
 ## pkg local_list ~/.octave_packages
 ## @end example
diff --git a/scripts/plot/pareto.m b/scripts/plot/pareto.m
--- a/scripts/plot/pareto.m
+++ b/scripts/plot/pareto.m
@@ -40,18 +40,18 @@
 ## absent, then the abscissa are assumed to be @code{1 : length (@var{x})}.
 ## @var{y} can be a string array, a cell array of strings or a numerical
 ## vector.
 ##
 ## An example of the use of @code{pareto} is
 ##
 ## @example
 ## @group
-## Cheese = @{"Cheddar", "Swiss", "Camembert", "Munster", "Stilton", @dots{}
-##    "Blue"@};
+## Cheese = @{"Cheddar", "Swiss", "Camembert", ...
+##           "Munster", "Stilton", "Blue"@};
 ## Sold = [105, 30, 70, 10, 15, 20];
 ## pareto(Sold, Cheese);
 ## @end group
 ## @end example
 ## @end deftypefn
 
 function h = pareto (varargin)
 
diff --git a/scripts/polynomial/compan.m b/scripts/polynomial/compan.m
--- a/scripts/polynomial/compan.m
+++ b/scripts/polynomial/compan.m
@@ -32,16 +32,17 @@
 ##      0    &     1    & \cdots &     0    &         0   \cr
 ##   \vdots  &   \vdots & \ddots &  \vdots  &      \vdots \cr
 ##      0    &     0    & \cdots &     1    &         0}\right].
 ## $$
 ## @end tex
 ## @end iftex
 ## @ifnottex
 ##
+## @c Set example in small font to prevent overfull line
 ## @smallexample
 ##      _                                                        _
 ##     |  -c(2)/c(1)   -c(3)/c(1)  @dots{}  -c(N)/c(1)  -c(N+1)/c(1)  |
 ##     |       1            0      @dots{}       0             0      |
 ##     |       0            1      @dots{}       0             0      |
 ## A = |       .            .   .            .             .      |
 ##     |       .            .       .        .             .      |
 ##     |       .            .           .    .             .      |
diff --git a/scripts/signal/arch_rnd.m b/scripts/signal/arch_rnd.m
--- a/scripts/signal/arch_rnd.m
+++ b/scripts/signal/arch_rnd.m
@@ -18,24 +18,26 @@
 ## <http://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
 ## @deftypefn {Function File} {} arch_rnd (@var{a}, @var{b}, @var{t})
 ## Simulate an ARCH sequence of length @var{t} with AR
 ## coefficients @var{b} and CH coefficients @var{a}.  I.e., the result
 ## @math{y(t)} follows the model
 ##
+## @c Set example in small font to prevent overfull line
 ## @smallexample
 ## y(t) = b(1) + b(2) * y(t-1) + @dots{} + b(lb) * y(t-lb+1) + e(t),
 ## @end smallexample
 ##
 ## @noindent
 ## where @math{e(t)}, given @var{y} up to time @math{t-1}, is
 ## @math{N(0, h(t))}, with
 ##
+## @c Set example in small font to prevent overfull line
 ## @smallexample
 ## h(t) = a(1) + a(2) * e(t-1)^2 + @dots{} + a(la) * e(t-la+1)^2
 ## @end smallexample
 ## @end deftypefn
 
 ## Author: KH <Kurt.Hornik@wu-wien.ac.at>
 ## Description: Simulate an ARCH process
 
diff --git a/src/DLD-FUNCTIONS/betainc.cc b/src/DLD-FUNCTIONS/betainc.cc
--- a/src/DLD-FUNCTIONS/betainc.cc
+++ b/src/DLD-FUNCTIONS/betainc.cc
@@ -41,16 +41,17 @@ Return the incomplete Beta function,\n\
 @tex\n\
 $$\n\
  \\beta (x, a, b) = B (a, b)^{-1} \\int_0^x t^{(a-z)} (1-t)^{(b-1)} dt.\n\
 $$\n\
 @end tex\n\
 @end iftex\n\
 @ifnottex\n\
 \n\
+@c Set example in small font to prevent overfull line\n\
 @smallexample\n\
                                       x\n\
                                      /\n\
 betainc (x, a, b) = beta (a, b)^(-1) | t^(a-1) (1-t)^(b-1) dt.\n\
                                      /\n\
                                   t=0\n\
 @end smallexample\n\
 @end ifnottex\n\
diff --git a/src/DLD-FUNCTIONS/filter.cc b/src/DLD-FUNCTIONS/filter.cc
--- a/src/DLD-FUNCTIONS/filter.cc
+++ b/src/DLD-FUNCTIONS/filter.cc
@@ -291,16 +291,17 @@ equation:\n\
 $$\n\
 \\sum_{k=0}^N a_{k+1} y_{n-k} = \\sum_{k=0}^M b_{k+1} x_{n-k}, \\qquad\n\
  1 \\le n \\le P\n\
 $$\n\
 @end tex\n\
 @end iftex\n\
 @ifnottex\n\
 \n\
+@c Set example in small font to prevent overfull line\n\
 @smallexample\n\
    N                   M\n\
   SUM a(k+1) y(n-k) = SUM b(k+1) x(n-k)      for 1<=n<=length(x)\n\
   k=0                 k=0\n\
 @end smallexample\n\
 @end ifnottex\n\
 \n\
 @noindent\n\
@@ -320,16 +321,17 @@ supplied.  An equivalent form of this eq
 $$\n\
 y_n = -\\sum_{k=1}^N c_{k+1} y_{n-k} + \\sum_{k=0}^M d_{k+1} x_{n-k}, \\qquad\n\
  1 \\le n \\le P\n\
 $$\n\
 @end tex\n\
 @end iftex\n\
 @ifnottex\n\
 \n\
+@c Set example in small font to prevent overfull line\n\
 @smallexample\n\
             N                   M\n\
   y(n) = - SUM c(k+1) y(n-k) + SUM d(k+1) x(n-k)  for 1<=n<=length(x)\n\
            k=1                 k=0\n\
 @end smallexample\n\
 @end ifnottex\n\
 \n\
 @noindent\n\
diff --git a/src/error.cc b/src/error.cc
--- a/src/error.cc
+++ b/src/error.cc
@@ -935,27 +935,27 @@ function g () h (); end\n\
 function h () nargin == 1 || error (\"nargin != 1\"); end\n\
 @end group\n\
 @end example\n\
 \n\
 @noindent\n\
 calling the function @code{f} will result in a list of messages that\n\
 can help you to quickly locate the exact location of the error:\n\
 \n\
-@smallexample\n\
+@example\n\
 @group\n\
 f ()\n\
 error: nargin != 1\n\
 error: called from:\n\
 error:   error at line -1, column -1\n\
 error:   h at line 1, column 27\n\
 error:   g at line 1, column 15\n\
 error:   f at line 1, column 15\n\
 @end group\n\
-@end smallexample\n\
+@end example\n\
 \n\
 If the error message ends in a new line character, Octave will print the\n\
 message but will not display any traceback messages as it returns\n\
 control to the top level.  For example, modifying the error message\n\
 in the previous example to end in a new line causes Octave to only print\n\
 a single message:\n\
 \n\
 @example\n\
diff --git a/src/load-save.cc b/src/load-save.cc
--- a/src/load-save.cc
+++ b/src/load-save.cc
@@ -1816,16 +1816,17 @@ Query or set the internal variable that 
 string used for the comment line written at the beginning of\n\
 text-format data files saved by Octave.  The format string is\n\
 passed to @code{strftime} and should begin with the character\n\
 @samp{#} and contain no newline characters.  If the value of\n\
 @code{save_header_format_string} is the empty string,\n\
 the header comment is omitted from text-format data files.  The\n\
 default value is\n\
 \n\
+@c Set example in small font to prevent overfull line\n\
 @smallexample\n\
 \"# Created by Octave VERSION, %a %b %d %H:%M:%S %Y %Z <USER@@HOST>\"\n\
 @end smallexample\n\
 @seealso{strftime}\n\
 @end deftypefn")
 {
   return SET_INTERNAL_VARIABLE (save_header_format_string);
 }
diff --git a/src/mappers.cc b/src/mappers.cc
--- a/src/mappers.cc
+++ b/src/mappers.cc
@@ -504,23 +504,23 @@ Computes the error function,\n\
 @tex\n\
 $$\n\
  {\\rm erf} (z) = {2 \\over \\sqrt{\\pi}}\\int_0^z e^{-t^2} dt\n\
 $$\n\
 @end tex\n\
 @end iftex\n\
 @ifnottex\n\
 \n\
-@smallexample\n\
+@example\n\
                          z\n\
                         /\n\
 erf (z) = (2/sqrt (pi)) | e^(-t^2) dt\n\
                         /\n\
                      t=0\n\
-@end smallexample\n\
+@end example\n\
 @end ifnottex\n\
 @seealso{erfc, erfinv}\n\
 @end deftypefn")
 {
   octave_value retval;
   if (args.length () == 1)
     retval = args(0).erf ();
   else
diff --git a/src/variables.cc b/src/variables.cc
--- a/src/variables.cc
+++ b/src/variables.cc
@@ -2221,16 +2221,17 @@ Left alignment.\n\
 @item r\n\
 Right alignment (this is the default).\n\
 @item c\n\
 Centered (may only be applied to command %s).\n\
 @end table\n\
 \n\
 A command is composed like this:\n\
 \n\
+@c Set example in small font to prevent overfull line\n\
 @smallexample\n\
 %[modifier]<command>[:size_of_parameter[:center-specific[:balance]]];\n\
 @end smallexample\n\
 \n\
 Command and modifier is already explained.  The @code{size_of_parameter}\n\
 parameter tells how many columns the parameter will need for printing.\n\
 The @code{center-specific} parameter may only be applied to command\n\
 @samp{%s}.\n\
